entity-server-api/src/main/java/org/terracotta/entity/PassiveServerEntity.java

```diff
@@ -36,9 +36,15 @@
    *  under a given key, the messages it creates and sends for synchronization will be executed in that key, on the passive,
    *  via this invoke() entry-point.</p>
    *
+   * @param clientDescriptor source instance from which the invocation originates.
+   * @param currentOrderedId current id this invoke is part of
+   * @param eldestOrderedId earliest active id for this client
    * @param message The message from a client or upstream active server
    */
-  void invoke(M message) throws EntityUserException;
+  void invoke(ClientDescriptor clientDescriptor,
```

Comment:
@myronkscott @cschanck yes that's what I wanted to know. Currently for the Ehcache entity we are creating the scheduled message with the original message's id and client id. Just wanted to confirm if that still needs to be done or I can directly use the ids coming in the invoke. And I got the answer. The client descriptor will be preserved and not the message ids.